### PR TITLE
Fix #3401: Introduced SplashScreenWelcomeMsg PlatformParameter

### LINKED ISSUE #3401: Introduce a demo PlatformParameter (Compile Time)
Introduce a test PlatformParameter to verify the Compile time Flag architecture.

SplashScreenToast

Target PR date: 6 July 2021
Target completion date: 11 July 2021

### PR DESCRIPTION
**Pull request by Arjupta**
## Explanation

Fixes #3401 
This PR introduces SplashScreenWelcomeMsg PlatformParameter for testing the implementation of Compile Time Platform Parameters. Also this PR is a replica of https://github.com/oppia/oppia-android/pull/3405

## Screenshots
Screenshot 2021-07-07 121030

| Test 1 | Test 2 |
|:-:|:-:|
|Screenshot_1625639975 | Screenshot_1625639982 |

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/java/org/oppia/android/app/splash/SplashActivityPresenter.kt
Comment (jcqli) [inline app/src/main/java/org/oppia/android/app/splash/SplashActivityPresenter.kt]: Maybe `processPlatformParameters` since "loading status" seems like a UI thing.

File: app/src/sharedTest/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (jcqli) [inline app/src/sharedTest/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt]: Write as // comment since we don't need a public KDoc.

**Review by jcqli — APPROVED** [review]
A few quick comments checking that this matches the original review, LGTM after those changes are made!

File: app/src/main/java/org/oppia/android/app/splash/SplashActivityPresenter.kt
Comment (Arjupta) [reply app/src/main/java/org/oppia/android/app/splash/SplashActivityPresenter.kt]: Done

File: app/src/sharedTest/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (Arjupta) [reply app/src/sharedTest/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt]: Done

**Review by vinitamurthi — APPROVED** [review]

File: app/src/sharedTest/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt:86]: Add espresso screenshot test result

**Review by anandwana001 — CHANGES_REQUESTED** [review]

File: app/src/sharedTest/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt
Comment (Arjupta) [reply app/src/sharedTest/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt:86]: They are already added in the PR comment @anandwana001

**Review by rt4914 — APPROVED** [review]
Approving for code owner files.

**Review by anandwana001 — APPROVED** [review]
I tried few things like decor view and putting more delay, but it might take more time so we are going with the run only on espresso for this test `testSplashTestActivity_updateEmptyDatabase_readDatabaseValues_checkWelcomeMsgIsVisible` .

**Comment by Arjupta** [comment]
@vinitamurthi assigning it back to you for merging

**Pull request merged**
